### PR TITLE
Add documentation page for CSI snapshots and clones

### PR DESCRIPTION
**Pull request by sdodsley**
Also, fix some markdown formatting issues whilst I'm editing the README files

### CODE REVIEW

**Comment by dnix101** [comment]
Great addition to the documentation especially on the usage of snapshots and clones and enabling featuregates.

**Comment by Pure-AdamuKaapan** [comment]
Looks good to me, the feature gates notes are especially helpful.

You accidentally got reviews from *both* my github accounts! 😁

File: docs/csi-snapshot-clones.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  storageClassName: pure
+  dataSource:
+    kind: VolumeSnapshot
+    name: volumesnapshot-1
+    apiGroup: snapshot.storage.k8s.io
+```
+
+#### Create a clone of a PVC
+
+Use the following YAML to create a clone called `clone-of-pure-claim` of the PVC `pure-claim`:
```
Comment (dsupure) [inline docs/csi-snapshot-clones.md]: It would be nice to mention `pure-claim` and  `clone-of-pure-claim` need to use the same storageclass.

Comment (sdodsley) [reply docs/csi-snapshot-clones.md]: Good point - done.

**Pull request merged**